src/models/lineChart.js

```diff
@@ -306,6 +351,30 @@ nv.models.lineChart = function() {
                 chart.update();
             });
 
+            lines.dispatch.on('elementMouseover.tooltip', function(evt) {
+                tooltip.data(evt).position(evt.pos).hidden(false);
+
+                if(showDistX){
+                    container.select('.nv-distributionX .nv-series-' + evt.seriesIndex + ' .nv-distx-' + evt.pointIndex)
+                        .attr('y1', evt.relativePos[1] - availableHeight);
+                }
+                if(showDistY){
+                    container.select('.nv-distributionY .nv-series-' + evt.seriesIndex + ' .nv-disty-' + evt.pointIndex)
+                        .attr('x2', evt.relativePos[0] + distX.size());
+                }
+            });
+
+            lines.dispatch.on('elementMouseout.tooltip', function(evt) {
+                tooltip.hidden(true);
+                if(showDistX){
+                    container.select('.nv-distributionX .nv-series-' + evt.seriesIndex + ' .nv-distx-' + evt.pointIndex)
+                        .attr('y1', 0);
+                }
+                if(showDistY){
+                    container.select('.nv-distributionY .nv-series-' + evt.seriesIndex + ' .nv-disty-' + evt.pointIndex)
+                        .attr('x2', distY.size());
+                }
+            });
```

Comment:
Why are you moving these functions here?   They'll get overwritten every chart.update() in this case.